k8s/client.go

```diff
@@ -87,7 +102,48 @@ func (c *client) GetIngresses() ([]*v1beta1.Ingress, error) {
 	return ingresses, nil
 }
 
+func (c *client) GetIngresses(selector *NamespaceSelector) ([]*v1beta1.Ingress, error) {
+	if !c.namespaceController.HasSynced() {
+		return nil, errors.New("namespaces haven't synced yet")
+	}
+
+	allIngresses, err := c.GetAllIngresses()
+	if err != nil {
+		return nil, err
+	}
+
+	filteredIngresses := []*v1beta1.Ingress{}
+	for _, ingress := range allIngresses {
+		if ingressInSupportedNamespace(ingress, selector, c.namespaceStore.List()) {
```

Comment:
It is the whole list, but my thinking about moving it to `controller.updateIngresses()` is that `updateIngresses()` is called on a watcher update for any of the three resource types being watched (ingresses, namespaces and services). Hence any filtering we do there would still happen at every single update.

We could have a different callback method for the namespace watcher, which performs this filtering though. Is it worth the extra complication though, since even on a namespace update we need to update Feed's IngressEntries.

Reply:
We should still reduce the complexity of the current fitlering implementation for the worst case scenario from:
`All ingresses* All namespaces`
to
`All namespaces + All ingresses*Selected namespaces`